src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/MarkJavaObjects.cs

```diff
@@ -264,24 +262,30 @@ void PreserveConstructors (TypeDefinition type, TypeDefinition helper)
 				PreserveMethod (type, ctor);
 		}
 
-		static bool IsImplementor (TypeDefinition type)
-		{
-			return type.Name.EndsWith ("Implementor", StringComparison.Ordinal) && type.Inherits ("Java.Lang.Object");
-		}
-
 		static bool IsUserType (TypeDefinition type)
 		{
 			return !MonoAndroidHelper.IsFrameworkAssembly (type.Module.Assembly.Name.Name + ".dll");
 		}
 
 		void PreserveImplementor (TypeDefinition type)
 		{
-			if (!type.HasMethods)
+			const string implementor = "Implementor";
+			if (!type.Name.EndsWith (implementor, StringComparison.Ordinal) || !type.Inherits ("Java.Lang.Object"))
+				return;
+			if (!type.HasInterfaces)
 				return;
 
-			foreach (MethodDefinition method in type.Methods)
-				if (method.Name.EndsWith ("Handler", StringComparison.Ordinal))
-					PreserveMethod (type, method);
+			var ifaceName = "I" + type.Name.Substring (0, type.Name.Length - implementor.Length);
```

Comment:
This logic is "not quite right"; it's focusing on the `TextWatcherImplementor` case, at the expense of what `generator` emits.  Consider:

```
% grep -r 'class.*Implementor : ' src/Mono.Android/obj/Debug/net7.0/android-33/mcw
src/Mono.Android/obj/Debug/net7.0/android-33/mcw/Android.Animation.LayoutTransition.cs:		internal sealed partial class ITransitionListenerImplementor : global::Java.Lang.Object, ITransitionListener {
src/Mono.Android/obj/Debug/net7.0/android-33/mcw/Android.Preferences.PreferenceManager.cs:		internal sealed partial class IOnActivityDestroyListenerImplementor : global::Java.Lang.Object, IOnActivityDestroyListener {
src/Mono.Android/obj/Debug/net7.0/android-33/mcw/Android.Preferences.PreferenceManager.cs:		internal sealed partial class IOnActivityResultListenerImplementor : global::Java.Lang.Object, IOnActivityResultListener {
src/Mono.Android/obj/Debug/net7.0/android-33/mcw/Android.Preferences.PreferenceManager.cs:		internal sealed partial class IOnActivityStopListenerImplementor : global::Java.Lang.Object, IOnActivityStopListener {
src/Mono.Android/obj/Debug/net7.0/android-33/mcw/Android.Media.Audiofx.Virtualizer.cs:		internal sealed partial class IOnParameterChangeListenerImplementor : global::Java.Lang.Object, IOnParameterChangeListener {
src/Mono.Android/obj/Debug/net7.0/android-33/mcw/Android.Nfc.NfcAdapter.cs:		internal sealed partial class IOnTagRemovedListenerImplementor : global::Java.Lang.Object, IOnTagRemovedListener {
src/Mono.Android/obj/Debug/net7.0/android-33/mcw/Android.Bluetooth.IBluetoothProfile.cs:	internal sealed partial class IBluetoothProfileServiceListenerImplementor : global::Java.Lang.Object, IBluetoothProfileServiceListener {
src/Mono.Android/obj/Debug/net7.0/android-33/mcw/Android.Renderscripts.Allocation.cs:		internal sealed partial class IOnBufferAvailableListenerImplementor : global::Java.Lang.Object, IOnBufferAvailableListener {
src/Mono.Android/obj/Debug/net7.0/android-33/mcw/Android.Content.ClipboardManager.cs:		internal sealed partial class IOnPrimaryClipChangedListenerImplementor : global::Java.Lang.Object, IOnPrimaryClipChangedListener {
src/Mono.Android/obj/Debug/net7.0/android-33/mcw/Android.Views.Animations.Animation.cs:		internal sealed partial class IAnimationListenerImplementor : global::Java.Lang.Object, IAnimationListener {
…
```

The *normal* pattern is that *just* `Implementor` is appended.  `TextWatcherImplementor` is the odd one out, because it is hand-written.

Thus, at minimum, instead of prefixing with `I` and removing the `Implementor` suffix, we should instead:

 1. Just remove the `Implementor` suffix, and
 2. Rename `TextWatcherImplementor` to `ITextWatcherImplementor`.

But…. Do we really need to do this check?  Why not just preserve *all* the interfaces that the `*Implementor` type implements?

```csharp
foreach (var iface in type.Interfaces) {
    Annotations.Mark (iface.InterfaceType.Resolve ());
}
```

99.9999% of the time this will be *precisely* what we want, as this is what `generator` emits.  The only "problem" is that we may preserve "too much", but that would only be for types which are hand-written, which is highly unlikely to be anyone other than *us*.

Reply:
I changed this to preserve all interfaces on types that end with the name `Implementor`.

Reply:
Hmm, this increases the size a lot:
```
Error: apkdiff: Assembly 'assemblies/Mono.Android.dll' size increase 225,930 is 174,730 bytes more than the threshold 51,200.
```
So perhaps this is marking "too many" interfaces? Like maybe it's preserving ones that aren't used at all?